processor/agentmetricsprocessor/agentmetricsprocessor_test.go

```diff
@@ -188,18 +184,15 @@ func (mb metricBuilder) addDoubleDataPoint(value float64, labels map[string]stri
 // assertEqual is required because Attribute & Label Maps are not sorted by default
 // and we don't provide any guarantees on the order of transformed metrics
 func assertEqual(t *testing.T, expected, actual pdata.Metrics) {
-	rmsAct := pdatautil.MetricsToInternalMetrics(actual).ResourceMetrics()
-	rmsExp := pdatautil.MetricsToInternalMetrics(expected).ResourceMetrics()
+	rmsAct := actual.ResourceMetrics()
+	rmsExp := expected.ResourceMetrics()
 	require.Equal(t, rmsExp.Len(), rmsAct.Len())
 	for i := 0; i < rmsAct.Len(); i++ {
 		rmAct := rmsAct.At(i)
 		rmExp := rmsExp.At(i)
 
 		// assert equality of resource attributes
-		assert.Equal(t, rmExp.Resource().IsNil(), rmAct.Resource().IsNil())
-		if !rmExp.Resource().IsNil() {
```

Comment:
Was there a specific case where resource was nil in the tests? Isn't that needed anymore?

Reply:
Resources are now initialized on creation and should never be nil. The `IsNil` function was removed.

See https://github.com/open-telemetry/opentelemetry-collector/pull/2005

Reply:
(I don't think there was a test case to cover this 😮 )